repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -105,21 +113,31 @@ class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMo
   def this(originalResponse: HttpServletResponse, headerMode: ResponseMode, bodyMode: ResponseMode) =
     this(originalResponse, headerMode, bodyMode, originalResponse.getOutputStream)
 
+  override def getStatus: Int = {
+    logger.trace("getStatus called")
+    statusCode
+  }
+
   override def setStatus(i: Int): Unit = {
-    if (isCommitted) {
-      throw new IllegalStateException("Cannot call sendError or setStatus after the response has been committed")
-    } else {
-      super.setStatus(i)
-      reason = None
-    }
+    logger.trace(s"setStatus called with status code: $i")
+    setStatusCommon(i)(originalResponse.setStatus(i))
   }
 
   override def setStatus(i: Int, s: String): Unit = {
+    logger.trace(s"setStatus called with status code: $i, message: $s")
+    setStatusCommon(i, s)(originalResponse.setStatus(i, s))
+  }
+
+  private def setStatusCommon(i: Int, s: String = null)(setStatusFunc: => Unit): Unit = {
```

Comment:
Don't really like this method name. I'm assuming you can't just further overload it.

Reply:
I could overload `setStatus` further, I suppose. Although I probably am not allowed to use currying if I do, because I expect that makes this method indistinguishable from the other `setStatus` methods. (Just tested, and that's totally true)

Interestingly, IntelliJ complains when I try to overload `setStatus` here for some reason. It wants me to use the `override` modifier, but this method clearly does not override any other method. And, in fact, the code only compiles *without* the `override` modifier.

So, would you be happier with this?:
```
override def setStatus(i: Int): Unit = {
    logger.trace(s"setStatus called with status code: $i")
    setStatus(i, setStatusFunc = originalResponse.setStatus(i))
  }

  override def setStatus(i: Int, s: String): Unit = {
    logger.trace(s"setStatus called with status code: $i, message: $s")
    setStatus(i, s, originalResponse.setStatus(i, s))
  }

  private def setStatus(i: Int, s: String = null, setStatusFunc: => Unit): Unit = {
    if (isCommitted) {
      throw new IllegalStateException("Cannot call setStatus after the response has been committed")
    }

    statusCode = i
    reason = Option(s)

    if (headerMode != ResponseMode.MUTABLE && bodyMode != ResponseMode.MUTABLE) {
      setStatusFunc
    }
  }
```

Or perhaps with something slightly less obscure:
```
override def setStatus(i: Int): Unit = {
    logger.trace(s"setStatus called with status code: $i")
    setStatus(originalResponse.setStatus(i), i)
  }

  override def setStatus(i: Int, s: String): Unit = {
    logger.trace(s"setStatus called with status code: $i, message: $s")
    setStatus(originalResponse.setStatus(i, s), i, s)
  }

  private def setStatus(setStatusFunc: => Unit, i: Int, s: String = null): Unit = {
    if (isCommitted) {
      throw new IllegalStateException("Cannot call setStatus after the response has been committed")
    }

    statusCode = i
    reason = Option(s)

    if (headerMode != ResponseMode.MUTABLE && bodyMode != ResponseMode.MUTABLE) {
      setStatusFunc
    }
  }
```

Both of those also work, but personally, I find what I currently have to be the clearest and most flexible expression of this method. I could also just change the name to something else.

Reply:
Yeah, i'm mostly ok with the implicit usage, i just don't like the name. maybe `doSetStatus` that's a sort of standard for this type of thing.

Reply:
Sure, I'll update the method names to follow that, then.

Oh, and just to be clear, no implicits are being used here, just multiple parameter lists. I know this is the pattern followed when implicits are used, but I wanted to call that out so that no one thinks there is magic happening in this case.